### PR TITLE
[onert] Introduce CategoricalCrossEntropy kernel

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit introduces Categorical Cross Entropy forward and backward kernels.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Draft: #11769 
Related issue: #11857

### CODE REVIEW

File: compute/cker/src/train/Loss.test.cc [hunk 1]
```diff
@@ -199,3 +278,180 @@ TEST(CKer_Operation, neg_LossMSEGrad)
...
+    nnfw::cker::Shape in_shape{1, 10};
+    nnfw::cker::Shape out_shape{1, 1};
+
+    std::vector<float> y_pred = {2.86E-12, 2.82E-13, 0.99999845, 2.36E-07, 2.91E-16,
+                                 2.10E-07, 1.69E-14, 1.21E-17,   1.08E-06, 6.23E-18};
+    std::vector<float> y_true = {0, 0, 0, 0, 0, 0, 0, 0, 0, 1};
+    std::vector<float> expected = {39.617155};
+
+    LossCCEVerifier<float> verifier(in_shape, out_shape);
+    verifier.verifyForward(y_pred, y_true, expected);
```
Comment (glistening) [inline compute/cker/src/train/Loss.test.cc:294]: I am not against this PR. But for future, I have some questions:

1. How did you prepare `y_pred` values?
2. How did you prepare expected loss value?

File: compute/cker/src/train/Loss.test.cc [hunk 2]
```diff
@@ -19,6 +19,85 @@
...
+
+    std::vector<T> output(_out_shape.FlatSize());
+    const int N = _in_shape.Dims(0);
+    const int D = _in_shape.FlatSize() / N;
+
+    nnfw::cker::train::CategoricalCrossEntropy(y_pred.data(), y_true.data(), output.data(), N, D);
+
+    for (int i = 0; i < output.size(); ++i)
+    {
+      EXPECT_FLOAT_EQ(output[i], expected[i]);
```
Comment (glistening) [inline compute/cker/src/train/Loss.test.cc:47]: As we use `EXPECT_FLOAT_EQ`, the input/expected/... needs to be selected carefully. What about leaving a comment "Don't be panic when it fails after kernel implementation or input is changed.".

File: compute/cker/src/train/Loss.test.cc (hunk 1 above)
Comment (jyoungyun) [reply compute/cker/src/train/Loss.test.cc:294]: The `y_pred` value is the softmax input value of the `fashion mnist` dataset.
Then, I manually calcuated the expected loss value using  https://github.com/Samsung/ONE/issues/11248#issuecomment-1807569062 formula. Unfortunately, the log output varies slightly depending on the library, environment, etc. I used python numpy api.

**Review by glistening — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]
LGTM
